provider/rosa_operator_roles_data_source.go

```diff
@@ -0,0 +1,230 @@
+	}
+
+	stsOperatorMap := make(map[string]*cmv1.STSOperator)
+	stsOperatorRolesList.Items().Each(func(stsCredentialRequest *cmv1.STSCredentialRequest) bool {
+		// TODO: check the MinVersion of the operator role
+		t.logger.Debug(ctx, "Operator name: %s, namespace %s, service account %s",
+			stsCredentialRequest.Operator().Name(),
+			stsCredentialRequest.Operator().Namespace(),
+			stsCredentialRequest.Operator().ServiceAccounts(),
+		)
+		// The key can't be stsCredentialRequest.Operator().Name() because of constants between
+		// the name of `ingress_operator_cloud_credentials` and `cloud_network_config_controller_cloud_credentials`
+		// both of them includes the same Name `cloud-credentials` and it cannot be fixed
+		stsOperatorMap[stsCredentialRequest.Operator().Namespace()] = stsCredentialRequest.Operator()
+		return true
+	})
+
+	get, err := t.clustersClient.Cluster(state.ClusterID.Value).Get().SendContext(ctx)
+	if err != nil {
+		response.Diagnostics.AddError(
+			"Can't find cluster",
+			fmt.Sprintf(
+				"Can't find cluster with identifier '%s': %v",
+				state.ClusterID.Value, err,
+			),
+		)
+		return
+	}
+	object := get.Body()
+	sts, ok := object.AWS().GetSTS()
+	if ok {
+		accountRolePrefix := DefaultAccountRolePrefix
+		if !state.AccountRolePrefix.Unknown && !state.AccountRolePrefix.Null {
+			accountRolePrefix = state.AccountRolePrefix.Value
+		}
+
+		// TODO: use the sts.OperatorRolePrefix() if not empty
+		// There is a bug in the return value of sts.OperatorRolePrefix() - it's always empty string
+		for _, operatorRole := range sts.OperatorIAMRoles() {
+			r := OperatorIAMRole{
+				Name: types.String{
+					Value: operatorRole.Name(),
+				},
+				Namespace: types.String{
+					Value: operatorRole.Namespace(),
+				},
+				RoleARN: types.String{
+					Value: operatorRole.RoleARN(),
+				},
+				RoleName: types.String{
+					Value: getRoleName(state.OperatorRolePrefix.Value, operatorRole.Namespace(), operatorRole.Name()),
+				},
+				PolicyName: types.String{
+					Value: getRoleName(accountRolePrefix, operatorRole.Namespace(), operatorRole.Name()),
+				},
+				ServiceAccounts: getServiceAccount(stsOperatorMap[operatorRole.Namespace()].ServiceAccounts(), operatorRole.Namespace()),
+			}
+			state.OperatorIAMRoles = append(state.OperatorIAMRoles, &r)
+		}
+	}
+	// Save the state:
+	diags = response.State.Set(ctx, state)
+	response.Diagnostics.Append(diags...)
+}
+
+// TODO: should be in a separate repo
+func getRoleName(prefix string, namespace string, name string) string {
+	roleName := fmt.Sprintf("%s-%s-%s", prefix, namespace, name)
+	if len(roleName) > 64 {
+		roleName = roleName[0:64]
+	}
+	return roleName
+}
+
+func getServiceAccount(serviceAccountArr []string, operatorNamespace string) types.List {
+	serviceAccounts := types.List{
+		ElemType: types.StringType,
+		Elems:    []attr.Value{},
+	}
+
+	for _, v := range serviceAccountArr {
+		serviceAccount := fmt.Sprintf(serviceAccountFmt, operatorNamespace, v)
```

Comment:
What if `operatorNamespace` is empty? should it have special behavior?

Reply:
This should be a bug in uhc-cluster-service (I'm getting it from the OCM API) should it will influence so many other places...